src/authTokenInterceptor.ts

```diff
@@ -182,33 +148,30 @@ export const authTokenInterceptor = ({
   return async (requestConfig: any): Promise<any> => {
     // Waiting for a fix in axios types
     // We need refresh token to do any authenticated requests
-    if (!getRefreshToken()) return requestConfig
-
-    // Queue the request if another refresh request is currently happening
-    if (isRefreshing) {
-      return new Promise((resolve, reject) => {
-        queue.push({ resolve, reject })
-      })
-        .then((token) => {
-          if (requestConfig.headers) {
-            requestConfig.headers[header] = `${headerPrefix}${token}`
-          }
-          return requestConfig
-        })
-        .catch(Promise.reject)
-    }
-
-    // Do refresh if needed
-    let accessToken
-    try {
-      accessToken = await refreshTokenIfNeeded(requestRefresh)
-      resolveQueue(accessToken)
-    } catch (error: unknown) {
-      if (error instanceof Error) {
-        declineQueue(error)
-        throw new Error(
-          `Unable to refresh access token for request due to token refresh error: ${error.message}`
-        )
+    if (!(await getRefreshToken())) return requestConfig
+
+    let accessToken = undefined
+
+    // Try to await a current request
+    if (currentlyRequestingPromise) accessToken = await currentlyRequestingPromise
+
+    if (!accessToken) {
```

Comment:
Maybe we can return early here for clarity

Reply:
Can't return here - needs to fall through to the code below to set the headers and and the config.